src/Interpreters/Squashing.cpp

```diff
@@ -11,93 +14,99 @@ namespace ErrorCodes
-            Chunk res_chunk = convertToChunk(std::move(chunks_to_merge_vec));
-            chunks_to_merge_vec.clear();
-            return res_chunk;
+            accumulated.add(std::move(input_chunk));
+            return convertToChunk(extract());
         }
 
         /// Return accumulated data (maybe it has small size) and place new block to accumulated data.
-        Chunk res_chunk = convertToChunk(std::move(chunks_to_merge_vec));
-        chunks_to_merge_vec.clear();
-        changeCurrentSize(input_chunk.getNumRows(), input_chunk.bytes());
-        chunks_to_merge_vec.push_back(std::move(input_chunk));
+        Chunk res_chunk = convertToChunk(extract());
+        accumulated.add(std::move(input_chunk));
         return res_chunk;
     }
 
     /// Accumulated block is already enough.
-    if (isEnoughSize(accumulated_size.rows, accumulated_size.bytes))
+    if (isEnoughSize())
     {
         /// Return accumulated data and place new block to accumulated data.
-        Chunk res_chunk = convertToChunk(std::move(chunks_to_merge_vec));
-        chunks_to_merge_vec.clear();
-        changeCurrentSize(input_chunk.getNumRows(), input_chunk.bytes());
-        chunks_to_merge_vec.push_back(std::move(input_chunk));
+        Chunk res_chunk = convertToChunk(extract());
+        accumulated.add(std::move(input_chunk));
         return res_chunk;
     }
 
     /// Pushing data into accumulating vector
-    expandCurrentSize(input_chunk.getNumRows(), input_chunk.bytes());
-    chunks_to_merge_vec.push_back(std::move(input_chunk));
+    accumulated.add(std::move(input_chunk));
 
     /// If accumulated data is big enough, we send it
-    if (isEnoughSize(accumulated_size.rows, accumulated_size.bytes))
-    {
-        Chunk res_chunk = convertToChunk(std::move(chunks_to_merge_vec));
-        changeCurrentSize(0, 0);
-        chunks_to_merge_vec.clear();
-        return res_chunk;
-    }
+    if (isEnoughSize())
+        return convertToChunk(extract());
+
     return {};
 }
 
-Chunk Squashing::convertToChunk(std::vector<Chunk> && chunks) const
+Chunk Squashing::convertToChunk(CurrentData && data) const
 {
-    if (chunks.empty())
+    LOG_INFO(getLogger("Squashing"), "convertToChunk {}", data.chunks.size());
+
+    if (data.chunks.empty())
         return {};
 
     auto info = std::make_shared<ChunksToSquash>();
-    info->chunks = std::move(chunks);
-
-    chunks.clear();
-
-    return Chunk(header.cloneEmptyColumns(), 0, info);
+    info->chunks = std::move(data.chunks);
+
+    // It is imortant that chunk is not empty, it has to have columns even if they are empty
+    // Sometimes there are could be no columns in header but not empty rows in chunks
+    // That happens when we intend to add defaults for the missing columns after
+    auto aggr_chunk = Chunk(header.getColumns(), header.columns() ? 0 : data.getRows());
+    aggr_chunk.getChunkInfos().add(std::move(info));
+    chassert(aggr_chunk);
```

Comment:
I have added that `chassert(aggr_chunk)` in oreder to check that `Squashing` produces chunks which converts to `bool` as `true`.
This has to be valid always, otherwise chunk might dropped in some processor.
Also chunk has to have the same columns as the processor header.

BUT. Sometime there is a chunks with `0` columns passing through pipeline. And it is valid situation.